patches/tModLoader/Terraria/ModLoader/ModSystem.cs

```diff
@@ -213,7 +214,14 @@ public abstract partial class ModSystem : ModType
 		public virtual void PostDrawInterface(SpriteBatch spriteBatch) { }
 
 		/// <summary>
-		/// Called while the fullscreen map is active. Allows custom drawing to the map.
+		/// Called right before map icon overlays are drawn. Use this hook to selectively hide existing <see cref="IMapLayer"/> or <see cref="ModMapLayer"/>
+		/// </summary>
+		/// <param name="layers"></param>
+		/// <param name="mapOverlayDrawContext"></param>
+		public virtual void PreDrawMapIconOverlay(List<IMapLayer> layers, MapOverlayDrawContext mapOverlayDrawContext) { }
```

Comment:
Sounds like this list has no reason to not be readonly. Also, shouldn't "Overlay" be plural?